### ISSUE TITLE
Harvesting a non-existent set processes forever

### ISSUE THREAD

**Issue by bkiahstroud** [harvard-lts/CURIOSity#164]

**Comment by bkiahstroud** [comment]
@VanessaVenti I don't see any existing code that handles this situation so I believe this is old behavior as opposed to a new bug caused by recent development. Given that, does adding the `Future Enhancements` label to this ticket make the most sense?

**Comment by VanessaVenti** [comment]
@bkiahstroud that's fine with me as long as there is a process for killing the job and that it doesn't cause other issues.

**Comment by bkiahstroud** [comment]
Process for killing a failed job: 

1. Login as an admin 
2. Navigate to `/delayed_job/overview` 
3. Click on "Failed Jobs" 
4. Click "Clear Failed Jobs" 

![Delayed Job Web 2022-08-26 at 10 31 40 AM](https://user-images.githubusercontent.com/32469930/186960650-c51f1542-b1b2-4b1a-b986-8be110ba6397.jpg)

**Comment by VanessaVenti** [comment]
Thanks @bkiahstroud! Tagging @phil-plencner-hl and @dl-maura so they see this to add to our documentation.

**Comment by dl-maura** [comment]
Hi @bkiahstroud We're adding this to our documentation, but is the mismatch status something that can be addressed? Like is there a way for it to understand it Failed since this can't be retried successfully ever? Or some indicator beyond 'oh I guess this has been open for a few hours but says successful'?

**Comment by bkiahstroud** [comment]
@dl-maura I'm sure a solution is possible, but I'm not sure what it would look like. And it would likely depend on what the desired behavior / UX would be for that specific situation. E.g. "Should the harvester job retry itself at all? If so, how many times? How long should it be able to run before being considered failed?" etc.

**Comment by VanessaVenti** [comment]
@bkiahstroud If it isn't able to find a setSpec because it doesn't exist I don't think it makes sense for the job to retry at all. Do you agree @dl-maura? Is there a way to first check if the setSpec exists and then give an error saying it was not found or just mark it as failed?

**Comment by dl-maura** [comment]
@VanessaVenti @bkiahstroud Yes I agree -- this is an error where it will never be successful because the setSpec is non-existent so it seems like it would be best if doesn't retry if it gets the 'doesn't exist' error:
![Screen Shot 2022-10-21 at 8 59 48 AM](https://user-images.githubusercontent.com/61880785/197201648-e9df3394-4bc3-4782-a280-b4ec8ce0190f.png)

If possible, that should just not retry, and the modal/stats should show as failed vs success, and if possible show that error like it should others in the modal. I don't know how hard it is to tell a job to auto clear so it may be that we will still need to do the delayed_job failure clear to end the time issue, but that way at least the person knows it failed, and why? @VanessaVenti what do you think?

**Comment by VanessaVenti** [comment]
Yes, agreed that all sounds good to me!

**Comment by bkiahstroud** [comment]
>I don't know how hard it is to tell a job to auto clear so it may be that we will still need to do the delayed_job failure clear to end the time issue, but that way at least the person knows it failed, and why?

@dl-maura not hard, but it does require understanding how the job and the job tracker interact with each other: 

```
"Job" --> Delayed::Job, handles actual processing, viewable in Delayed Job dashboard 
"Job Tracker" --> Spotlight::JobTracker, reports information about Job in app UI, viewable in indexing status modal 
```

If an error is thrown during the Job and **is not** captured (`rescue`d), the Job itself with fail (viewable in Delayed Job Dashboard) and retry. 

If an error is thrown during the Job and **is** captured (`rescue`d), the error information can be logged on the JobTracker and the JobTracker can be marked as failed (viewable in modal). If the error is not re-raised, the Job will not fail and thus won't retry.

**Comment by dl-maura** [comment]
ideal: if the job error is for non-existent set “No records match criteria”, clear the failed job automatically and mark as failed in the dashboard view, not "successful"  (which currently happens because the set doesn't exist so 0/0 processed, success)

**Comment by dl-maura** [comment]
Whether this is high priority will depend on the set spec name ticket() which @dl-maura and @phil-plencner-hl are working on on Friday

**Comment by dl-maura** [comment]
We were able to add set spec name so we're going to move this to like 'future' work
